test/foundry/invariants/IPAccount.t.sol

```diff
@@ -0,0 +1,281 @@
+
+        uint256 recorded = recorder.recorded();
+
+        assertGe(recorded, 0, "recorded");
+        assertEq(_chainId, block.chainid, "chainId");
+        assertEq(_tokenContract, address(mockNFT), "tokenContract");
+        assertEq(_owner, address(harness), "owner");
+    }
+
+    /// @notice Invariant to check that if a tx is executed, the state shall be updated
+    function invariant_permissionedStateUpdate() public {
+        if (recorder.recorded() > 0) {
+            bytes32 _state = ipAccount.state();
+            assertNotEq(_state, state, "state");
+        }
+    }
+}
+
+// contract IPAccountReplayHarness is BaseTest {
+//     bytes public signature;
+//     uint256 public value = 0;
+//     bytes public data = abi.encodeWithSignature("lol()");
+//     address public signer;
+//     uint256 public deadline = block.timestamp + 100;
+//     IPAccountImpl public ipAccount;
+//     Recorder public recorder;
+
+//     constructor() {
+//         recorder = new Recorder();
+
+//         // create a private key
+//         (address _signer, uint256 key) = makeAddrAndKey("signer");
+
+//         signer = _signer;
+
+//         vm.startPrank(address(_signer));
+//         mockNFT.mintId(address(_signer), 300);
+//         address _ipAccount = ipAssetRegistry.register(block.chainid, address(mockNFT), 300);
+//         assertTrue(ipAssetRegistry.isRegistered(_ipAccount));
+//         vm.stopPrank();
+
+//         ipAccount = IPAccountImpl(payable(_ipAccount));
+
+//         bytes32 digest = MessageHashUtils.toTypedDataHash(
+//             MetaTx.calculateDomainSeparator(),
+//             MetaTx.getExecuteStructHash(
+//                 MetaTx.Execute({
+//                     to: address(recorder),
+//                     value: value,
+//                     data: data,
+//                     nonce: ipAccount.state(),
+//                     deadline: deadline
+//                 })
+//             )
+//         );
+//         (uint8 v, bytes32 r, bytes32 s) = vm.sign(key, digest);
+//         signature = abi.encodePacked(v, r, s);
+//         ipAccount.executeWithSig(address(recorder), value, data, signer, deadline, signature);
+//     }
+
+//     function executeWithSigReplay() external {
+//         ipAccount.executeWithSig(address(recorder), value, data, signer, deadline, signature);
+//     }
+// }
+
+// /// @notice Signature replay attack invariants for IPAccount
+// contract IPAccountReplayInvariants is BaseTest {
+//     IPAccountReplayHarness public harness;
+//     Recorder public recorder;
+
+//     function setUp() public override {
+//         super.setUp();
+//         harness = new IPAccountReplayHarness();
+//         recorder = harness.recorder();
+//         targetContract(address(harness));
+//     }
+
+//     function invariant_replay() public {
+//         uint256 recorded = recorder.recorded();
+//         assertEq(recorded, 1, "recorded");
+//     }
+// }
```

Comment:
If the commented code is not used now, then should be removed.

Reply:
Removed.